packages/browser/src/core/task/task-group.ts

```diff
@@ -0,0 +1,31 @@
+import { isThenable } from '../../lib/is-thenable'
+
+export type TaskGroup = {
+  done: () => Promise<void>
+  run: <Operation extends (...args: any[]) => any>(
+    op: Operation
+  ) => ReturnType<Operation>
+}
+
+export const createTaskGroup = (): TaskGroup => {
+  let taskCompletionPromise: Promise<void>
+  let resolvePromise: () => void
+  let count = 0
+
+  return {
+    done: () => taskCompletionPromise,
+    run: (op) => {
+      const returnValue = op()
+
+      if (isThenable(returnValue)) {
+        if (++count === 1) {
+          taskCompletionPromise = new Promise((res) => (resolvePromise = res))
+        }
+
+        returnValue.finally(() => --count === 0 && resolvePromise())
```

Comment:
Can you confirm that the polyfill we use in the standalone version (CDN version) includes Promise.finally? I think this is the 1st time we'd be using Promise.finally outside npm-specific code.

Reply:
I didn't find any polyfilling at all anywhere. Tried to follow our build pipeline till the end, nothing, and no reference to 'core-js` either.

Reply:
We only polyfill for the CDN version but we load a file here depending on if we determine we should polyfill - https://github.com/segmentio/analytics-next/blob/aa3c901acd77597aa3c9bcb992369a39ef860919/packages/browser/src/browser/standalone.ts#L57-L63

Reply:
The polyfill works and makes `finally` work as expected. Tested in IE11 which has no Promise support at all, but once the polyfill script is executed in the console, Promise and `finally` work nominally.

Reply:
Sweet, thanks!